src/MongoDB.Driver.Core/Core/Operations/RetryableDeleteCommandOperation.cs

```diff
@@ -97,41 +89,31 @@ protected override BsonDocument CreateCommand(ConnectionDescription connectionDe
                 }
             }
 
-            var batchSerializer = CreateBatchSerializer(connectionDescription, attempt);
-            var batchWrapper = new BsonDocumentWrapper(_deletes, batchSerializer);
-
-            BsonDocument writeConcernWrapper = null;
-            if (WriteConcernFunc != null)
-            {
-                var writeConcernSerializer = new DelayedEvaluationWriteConcernSerializer();
-                writeConcernWrapper = new BsonDocumentWrapper(WriteConcernFunc, writeConcernSerializer);
-            }
-
             return new BsonDocument
             {
                 { "delete", _collectionNamespace.CollectionName },
-                { "ordered", _isOrdered },
-                { "txnNumber", () => transactionNumber.Value, transactionNumber.HasValue },
-                { "deletes", new BsonArray { batchWrapper } },
-                { "writeConcern", writeConcernWrapper, writeConcernWrapper != null }
+                { "ordered", IsOrdered },
+                { "writeConcern", () => WriteConcern.ToBsonDocument(), WriteConcern != null && !WriteConcern.IsServerDefault },
+                { "txnNumber", () => transactionNumber.Value, transactionNumber.HasValue }
             };
         }
 
-        // private methods
-        private IBsonSerializer<BatchableSource<DeleteRequest>> CreateBatchSerializer(ConnectionDescription connectionDescription, int attempt)
+        /// <inheritdoc />
+        protected override IEnumerable<Type1CommandMessageSection> CreateCommandPayloads(IChannelHandle channel, int attempt)
         {
+            BatchableSource<DeleteRequest> deletes;
             if (attempt == 1)
             {
-                var maxBatchCount = Math.Min(MaxBatchCount ?? int.MaxValue, connectionDescription.MaxBatchCount);
-                var maxItemSize = connectionDescription.MaxWireDocumentSize;
-                var maxBatchSize = connectionDescription.MaxDocumentSize;
-                return new SizeLimitingBatchableSourceSerializer<DeleteRequest>(DeleteRequestSerializer.Instance, NoOpElementNameValidator.Instance, maxBatchCount, maxItemSize, maxBatchSize);
+                deletes = _deletes;
             }
             else
             {
-                var count = _deletes.ProcessedCount; // as set by the first attempt
-                return new FixedCountBatchableSourceSerializer<DeleteRequest>(DeleteRequestSerializer.Instance, NoOpElementNameValidator.Instance, count);
+                deletes = new BatchableSource<DeleteRequest>(_deletes.Items, _deletes.Offset, _deletes.ProcessedCount, canBeSplit: false);
```

Comment:
Attempts after the first have to send the same number of requests as the first attempt. That is, they can't be split.